torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -2110,9 +2110,14 @@ class ShapePropagator : public PropertyPropBase {
       sizes[dim] = index->sizes()[0].value();
       node->output()->setType(ten->withSizes(sizes));
       return true;
-    } else if (node->matches(
-                   "aten::chunk(Tensor self, int chunks, int dim) -> Tensor[]",
-                   /*const_inputs=*/{attr::chunks, attr::dim})) {
+    } else if (
+        node->matches(
+            "aten::chunk(Tensor self, int chunks, int dim, bool redistribute, bool drop_remainder) -> Tensor[]",
+            /*const_inputs=*/
+            {attr::chunks,
+             attr::dim,
+             attr::redistribute,
+             attr::drop_remainder})) {
```

Comment:
The algorithm here seems incorrect to me: the sizes for all chunks (except last) is set to 

`input_type->sizes()[dim] / chunks`, 

and the size of the last chunk is set to 

`input_type->sizes()[dim] % chunks`.

This is incorrect, if sizes[dim] is 5 and chunks 2, the sizes of the outputs will be 2 and 1.
Can anyone give me some suggestions what to do here ? Thank you

Reply:
I can totally change the logic, my comment was to point out the current existing logic doesn't seem correct to me (and it was different from the current logic in torch.chunk). I don't really understand how the difference in the behaviours went unnoticed.

Is there a way to adjust the number of node->outputs() ? The correct behavior requires computing the number of outputs based on redistribute & drop_remainder (see https://github.com/pytorch/pytorch/issues/60531#issuecomment-870986104).

Thank you !

Reply:
Sorry, I thought you added the drop_remainder feature. Authors from Pytorch may know the answer. We will update coordinately once we have the solution. Thanks!

Reply:
I added the drop_remainder feature, but the existing code, before my changes, doesn't look right to me :)